src/linalg.jl

```diff
@@ -86,3 +86,30 @@ function Base.getproperty(C::Cholesky{T, <:BlockDiagonal{T}}, x::Symbol) where T
     end
     return BlockDiagonal(map(f, blocks(B)))
 end
+
+# 3-Argument mul!
+LinearAlgebra.mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal) = _mul!(C, A, B)
+
+if VERSION ≥ v"1.3"
+    LinearAlgebra.mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal, α::Number, β::Number) = _mul!(C, A, B, α, β)
```

Comment:
style: to avoid the very long line
```suggestion
    function LinearAlgebra.mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal, α::Number, β::Number)
        return _mul!(C, A, B, α, β)
    end
```